### PR TITLE
renderer.on_finish output after aggregation

### PR DESCRIPTION
**Pull request by pierky**
If I'm right the `on_finish` method of renderer was no more considered. I added it after aggregation output.

### CODE REVIEW

**Comment by astrikos** [comment]
oh yes, you are totally right! From what I see on_start is also left out with the new way. Maybe you can add that as well for the PR?
Also, while I was writing it I was thinking maybe the payload should be an attribute of renderer class and not the command as it is now. What do you think? @danielquinn ?

**Comment by danielquinn** [comment]
I was thinking something similar actually, but I'd want to go a step further and make it so that renderer.payload isn't a string but an iterable that's rendered with line-breaks in between.  That'd let us return a single string, a list, or a generator.

File: ripe/atlas/tools/commands/report.py [hunk 1]
```diff
@@ -161,4 +165,6 @@ def multi_level_render(self, aggregation_data, indent=""):
         elif isinstance(aggregation_data, list):
 
             for index, data in enumerate(aggregation_data):
-                self.payload = "{}{} {}".format(self.payload, indent, self.renderer.on_result(data))
+                res = self.renderer.on_result(data)
+                if res:
+                    self.payload = "{}{} {}".format(self.payload, indent, self.renderer.on_result(data))
```
Comment (danielquinn) [inline ripe/atlas/tools/commands/report.py]: You're calling `self.renderer.on_result()` twice here.  Once to check if there's any output, and again to append it to the payload.  I'm not sure what the value of this bit is.

The restoration of the `on_start()` and `on_finish()` bits is awesome though.  Thanks for catching that.

Comment (pierky) [inline ripe/atlas/tools/commands/report.py]: Forgot to replace the 2nd call with `res`. Fixed now.

**Pull request merged**